acceptance/src/data.rs

```diff
@@ -127,8 +127,8 @@ pub async fn assert_scan_data(engine: Arc<dyn Engine>, test_case: &TestCaseInfo)
     let mut schema = None;
     let batches: Vec<RecordBatch> = scan
         .execute(engine)?
-        .into_iter()
         .map(|res| {
+            let res = res.unwrap();
```

Comment:
I'm having to unwrap here despite the test returning a result. This is because the iterator must output `RecordBatch` I could either:
1. Keep the unwrap
2. `map_ok` then `try_collect` if we only want to bubble up the `execute` errors.
3. Changes all the `unwrap`s to return a result, then `try_collect` at the end.

Reply:
Similar stuff is going on in the golden_tables test below too.

Reply:
I think the "right" way is option 3 where we use `try_collect` at the end. If that's proving difficult, just `unwrap`, it's a unit test so it's okay to panic.